docs/tutorials/quick-start/quick-start-part-4.md

```diff
@@ -0,0 +1,808 @@
+
+:::info
+
+When using Immer, you can either "mutate" an existing state object, or return a new state value yourself, but not both at the same time. See the Immer docs guides on [Pitfalls](https://immerjs.github.io/immer/docs/pitfalls) and [Returning New Data](https://immerjs.github.io/immer/docs/return) for more details.
+
+:::
+
+Our last step is to update the `<ReactionButtons>` component to dispatch the `reactionAdded` action when the user clicks a button:
+
+```jsx title="features/posts/ReactionButtons.jsx"
+import React from 'react'
+// highlight-start
+import { useDispatch } from 'react-redux'
+
+import { reactionAdded } from './postsSlice'
+// highlight-end
+
+const reactionEmoji = {
+  thumbsUp: '👍',
+  hooray: '🎉',
+  heart: '❤️',
+  rocket: '🚀',
+  eyes: '👀'
+}
+
+export const ReactionButtons = ({ post }) => {
+  // highlight-next-line
+  const dispatch = useDispatch()
+
+  const reactionButtons = Object.entries(reactionEmoji).map(([name, emoji]) => {
+    return (
+      <button
+        key={name}
+        type="button"
+        className="muted-button reaction-button"
+        // highlight-start
+        onClick={() =>
+          dispatch(reactionAdded({ postId: post.id, reaction: name }))
+        }
+        // highlight-end
+      >
+        {emoji} {post.reactions[name]}
+      </button>
+    )
+  })
+
+  return <div>{reactionButtons}</div>
+}
+```
+
+Now, every time we click a reaction button, the counter should increment. If we browse around to different parts of the app, we should see the correct counter values displayed any time we look at this post, even if we click a reaction button in the `<PostsList>` and then look at the post by itself on the `<SinglePostPage>`.
+
+## What You've Learned
+
+Here's what our app looks like after all these changes:
+
+<iframe
+  class="codesandbox"
+  src="https://codesandbox.io/embed/github/markerikson/redux-quickstart-example-app/tree/checkpoint-2-reactionButtons/?fontsize=14&hidenavigation=1&theme=dark"
+  title="redux-quick-start-example-app"
+  allow="geolocation; microphone; camera; midi; vr; accelerometer; gyroscope; payment; ambient-light-sensor; encrypted-media; usb"
+  sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"
+></iframe>
+
+It's actually starting to look more useful and interesting!
+
+We've covered a lot of information and concepts in this section. Let's recap the important things to remember:
+
+:::tip
+
+- **Any React component can use data from the Redux store as needed**
+  - Any component can read any data that is in the Redux store
+  - Multiple components can read the same data, even at the same time
+  - Components should try to return just the data they need from selectors
```

Comment:
I didn't quite get it. 
Does it mean that components should try to extract just the data they need via selectors, not the whole state of the app?